utils.go

```diff
@@ -26,7 +27,10 @@ func isNil[T comparable](impl T) bool {
 func (this *Container) clearAll() {
 	this.resolvers = make(map[typeID][]serviceResolver)
 	this.aliases = make(map[pointerTypeName][]pointerTypeName)
-	this.isBuilt = false
+	this.isSealed = false
+	this.DisableValidation = false
+	lastContainerID = atomic.Int32{}
```

Comment:
these 2 lines will cause imprevisible behavior.

when I call `con.clearAll()`, I did not expect it will impact other containers or future created containers.

IMO `con.clearAll()` should impact only `con`

if for some reason you really want to reset the lastContainerID like this, then I suggest you would do it in another global function such as `resetContainersIDCounter()`. 

Anyway I'm curious about what was the motivation to reset this counter?

Reply:
example of the imprevisible behavior:

```
func Test() {
  con1 = NewContainer()
  con1.RegisterSomething()
  con1.clearAll() //no change idea
  con1.RegisterSomethingElse()

  con2 = NewContainer()
  con2.Register()
```

I did not expect that `con1.clearAll()` would have impact on `con2`, and make it got the same ID as `con1`, if the test continue there will be conflict between `con1`  and `con2`
}

Reply:
IMO `con.clearAll()` should impact only `con`

if for some reason you really want to reset the lastContainerID like this, then I suggest you would do it in another global function such as `resetContainersIDCounter()`. 

Anyway I'm curious about what was the motivation to reset this counter?